src/app/admin/usersTable/UsersTable.tsx

```diff
@@ -224,10 +226,20 @@ const UsersTable: React.FC = () => {
                     ${error.logId}`);
                         break;
                 }
-            } else {
-                setUsers(data.userData);
-                setFilteredUsersCount(data.count);
+                return;
             }
+            setUsers(data.userData);
+            setFilteredUsersCount(data.count);
+
+            const { data: currentUser, error: currentUserError } = await getCurrentUser();
+            if (currentUserError) {
+                setErrorMessage(
+                    `Error occured when fetching current user: ${currentUserError.type}, Log ID: ${currentUserError.logId}`
+                );
+                return;
```

Comment:
Same here - if there's any errors in getCurrentUser, we return from fetchAndDisplayUserData without setting isLoading to false etc. (line 244 & 245)?

Reply:
Also right, have fixed